regulations/static/regulations/js/source/views/main/child-view.js

```diff
@@ -81,54 +81,42 @@ var ChildView = Backbone.View.extend({
-
-                    if (typeof window.history !== 'undefined' && typeof window.history.replaceState !== 'undefined') {
-                        // update hash in url
-                        window.history.replaceState(
-                            null,
-                            null,
-                            window.location.origin + window.location.pathname + window.location.search + '#' + this.activeSection
-                        );
-                    }
-
-                    return;
-                }
+      $.each(this.$sections, function(idx, $section) {
+        if ($section.offset().top + this.$contentHeader.height() >= $(window).scrollTop()) {
+          if (_.isEmpty(this.activeSection) || (this.activeSection !== $section.id)) {
+            this.activeSection = $section[0].id;
+            this.$activeSection = $($section[0]);
+            // **Event** trigger active section change
+            HeaderEvents.trigger('section:open', this.$activeSection.data('label'));
+            DrawerEvents.trigger('section:open', this.$activeSection.data('toc-id') || this.id);
+            MainEvents.trigger('paragraph:active', this.activeSection);
+
+            if (typeof window.history !== 'undefined' && typeof window.history.replaceState !== 'undefined') {
+              // update hash in url
+              window.history.replaceState(
+                null,
+                null,
+                window.location.origin + window.location.pathname + window.location.search + '#' + this.activeSection
+              );
             }
-        }
 
-        return this;
+            return false;
+          }
+        }
+      }.bind(this));
     },
 
     updateWayfinding: function() {
-        var i, len;
-
-        // cache all sections in the DOM eligible to be the active section
-        // also cache some jQobjs that we will refer to frequently
-        this.$contentHeader = this.$contentHeader || $('header.reg-header');
-
-        // sections that are eligible for being the active section
-        this.$contentContainer = $('#' + this.id).find('li[id], .reg-section, .appendix-section, .supplement-section');
-
-        // cache jQobjs of each reg section
-        len = this.$contentContainer.length;
-
-        // short term solution: sometimes, back buttoning on diffs, this.$sections undefined. why?
-        this.$sections = this.$sections || {};
-
-        for (i = 0; i < len; i++) {
-            this.$sections[i] = $(this.$contentContainer[i]);
-        }
+      // cache all sections in the DOM eligible to be the active section
+      // also cache some jQobjs that we will refer to frequently
+      this.$contentHeader = this.$contentHeader || $('header.reg-header');
+
+      // sections that are eligible for being the active section
+      this.$sections = this.$el
+        .find('li[id], .reg-section, .appendix-section, .supplement-section')
+        .map(function(idx, elm) {
+          return $(elm);
+        });
```

Comment:
The return value of the `map` should be a jQuery object, not a native array. Anyway, the purpose of the `map` is to cache the jQuerified elements--if we just use the match, iterating over the results gives us a list of native DOM elements, when our code wants jQuerified elements instead.